q2cli/tools.py

```diff
@@ -145,6 +143,62 @@ def peek(path):
         click.secho(metadata.format)
 
 
+@tools.command('inspect-metadata',
+               short_help='Inpect columns available in metadata.')
+@q2cli.option('--tsv/--no-tsv', default=False)
+@click.argument('path', type=click.Path(exists=True, file_okay=True,
+                                        dir_okay=False, readable=True))
+@q2cli.util.pretty_failure(traceback=None)
+def inspect_metadata(path, tsv, failure):
+    import qiime2
+    import qiime2.sdk
+
+    try:
+        artifact = qiime2.sdk.Result.load(path)
+    except Exception:
+        metadata = qiime2.Metadata.load(path)
+    else:
+        if isinstance(artifact, qiime2.Visualization):
+            raise Exception("Visualizations do not have metadata.")
+        elif artifact.has_metadata():
+            metadata = artifact.view(qiime2.Metadata)
+        else:
+            raise Exception("Artifact of %r does not have metadata."
```

Comment:
similar comment as above:

"Artifact %r cannot be viewed as QIIME 2 Metadata."

I think using the Artifact's repr (instead of its type) gives the user a little more info because it includes the UUID and type.

Reply:
I tried that initially and thought it looked a little strange seeing its python repr "bleed through" the command line

Reply:
with variadic input, I think I'll include the path and UUID in prose since it's no longer clear which one is responsible for the issue

Reply:
:+1: , that's a similar approach as MetadataHandler